ext/search_kit/Civi/Search/Admin.php

```diff
@@ -343,15 +358,15 @@ private static function getReference(string $fieldName, array $references) {
   }
 
   /**
-   * Boilerplate join clause
+   * Fill in boilerplate join clause with supplied values.
    *
    * @param string $nearCol
    * @param string $farCol
-   * @param string $targetTable
+   * @param string|null $targetTable
    * @param string|null $dynamicCol
    * @return array[]
    */
-  private static function getJoinConditions($nearCol, $farCol, $targetTable = NULL, $dynamicCol = NULL) {
+  private static function getJoinConditions(string $nearCol, string $farCol, string $targetTable = NULL, string $dynamicCol = NULL):array {
```

Comment:
I was wondering but it seems that if you have a default value, then you don't need to have `?string`, even in php8.

For return type hints, I think the standard is a space after the colon, at least that's what php.net uses, and I've seen elsewhere in civi. But there's a preexisting mix.